contract/contracts/selling-lock/src/entry.rs

```diff
@@ -1,39 +1,144 @@
-// Import from `core` instead of from `std` since we are in no-std mode
-use core::result::Result;
-
-// Import heap related library from `alloc`
-// https://doc.rust-lang.org/alloc/index.html
-use alloc::{vec, vec::Vec};
-
-// Import CKB syscalls and structures
-// https://nervosnetwork.github.io/ckb-std/riscv64imac-unknown-none-elf/doc/ckb_std/index.html
+use core::{result::Result};
+use alloc::{vec::Vec};
 use ckb_std::{
     debug,
-    high_level::{load_script, load_tx_hash},
-    ckb_types::{bytes::Bytes, prelude::*},
+    ckb_constants::Source,
+    high_level::{load_script, load_cell_lock, load_cell, QueryIter},
+    ckb_types::{bytes::Bytes, prelude::*, packed::*},
 };
-
 use crate::error::Error;
 
-pub fn main() -> Result<(), Error> {
-    // remove below examples and write your code here
+const PRICE_LEN: usize = 8;
+struct SellingInfo {
+    // The owner of the selling lock
+    owner_lock: Script,
+    // minimal capacity should pay to the owner
+    #[allow(dead_code)]
+    price: u64,
+}
 
-    let script = load_script()?;
+fn load_selling_info() -> Result<SellingInfo, Error> {
+    let script: Script = load_script()?;
     let args: Bytes = script.args().unpack();
-    debug!("script args is {:?}", args);
+    let price_start_point = args.len() - PRICE_LEN;
+    let mut price_buf = [0u8; PRICE_LEN];
+    let raw_args = args.as_ref();
+    let owner_lock = ScriptBuilder::default()
+        .code_hash(Byte32::new_unchecked(Bytes::from(raw_args[0..32].to_vec())))
+        .hash_type(Byte::from(raw_args[32]))
+        .args(Bytes::from(raw_args[33..price_start_point].to_vec()).pack())
+        .build();
+    price_buf.copy_from_slice(raw_args[price_start_point..].as_ref());
+    let price = u64::from_le_bytes(price_buf);
+    Ok(SellingInfo{owner_lock, price})
+}
 
+fn is_owner_of_selling_info(other_lock: &Script, selling_info: &SellingInfo) ->  bool {
+    let owner_lock = &selling_info.owner_lock;
+    other_lock.code_hash().as_reader().raw_data()[..] == owner_lock.code_hash().as_reader().raw_data()[..] &&
```

Comment:
https://docs.rs/ckb-types/0.101.1/ckb_types/packed/struct.Script.html#method.eq

maybe `other_lock.eq(selling_info.owner_lock)` is fine